### PR TITLE
feat(asset-events-panel): add render override prop for metadata

### PR DESCRIPTION
**Pull request by annitya**

### CODE REVIEW

File: src/components/AssetEventsPanel/interfaces.ts [hunk 1]
```diff
@@ -65,6 +66,16 @@ export interface AssetEventsPanelThemeProps {
...
 }
 
+export interface EventAddonsProp extends CogniteEvent {
+  typeAndSubtype: React.ReactNode;
+  start: string;
+  end: string;
+}
+
+export interface RenderMetadataProps {
+  renderEventMetadata: (event: EventAddonsProp) => React.ReactNode;
```
Comment (mrtraser) [inline src/components/AssetEventsPanel/interfaces.ts:76]: It should be optional?

**Review by mrtraser — CHANGES_REQUESTED** [review]
This is a good one!

To merge this into master you should add:
- tests for new property
- story for the new use case

File: src/components/AssetEventsPanel/interfaces.ts (hunk 1 above)
Comment (annitya) [reply src/components/AssetEventsPanel/interfaces.ts:76]: Agreed!

**Comment by annitya** [comment]
Updated all the things! Thanks for getting back too me so soon btw :)

File: src/components/AssetEventsPanel/components/AssetEventsPanelPure.test.tsx [hunk 2]
```diff
@@ -28,6 +30,42 @@ describe('AssetEventsPanel', () => {
...
   });
+
+  it('should allow custom rendering of metadata', done => {
+    const assetEvents = fakeEvents;
+
+    const eventCssId = (event: CogniteEvent) => `pirate-${event.id}`;
+
+    const eventMessage = (event: CogniteEvent) => {
+      const source = (event.metadata && event.metadata.source) || '';
+      return `${source}: Look at me... Look at me! I'm the captain now!`;
```
Comment (mrtraser) [inline src/components/AssetEventsPanel/components/AssetEventsPanelPure.test.tsx:41]: Can we have some `lorem ipsum` instead?)

Comment (annitya) [reply src/components/AssetEventsPanel/components/AssetEventsPanelPure.test.tsx]: I have ipsum'ed the lorem!

**Comment by polomani** [comment]
@annitya I've added you as a collaborator. could you re-open the PR, but from this repo this time? so we can see the ci results

**Comment by annitya** [comment]
[Build](https://github.com/cognitedata/gearbox.js/pull/643) passing.

**Comment by polomani** [comment]
closing in favour of https://github.com/cognitedata/gearbox.js/pull/643https://github.com/cognitedata/gearbox.js/pull/643

**Pull request closed without merging**